urid/src/mapper.rs

```diff
@@ -53,12 +114,50 @@ pub trait URIDMapper {
     /// they might need at initialization time. See the `URIDCache` for more information on how to
     /// achieve this.
     fn unmap(&self, urid: URID) -> Option<&Uri>;
+
+    /// Unsafe wrapper of the `map` method, used by the feature interface.
+    unsafe extern "C" fn extern_unmap(
+        handle: crate::sys::LV2_URID_Map_Handle,
+        urid: crate::sys::LV2_URID,
+    ) -> *const c_char {
+        let urid = match URID::new(urid) {
+            None => return ::std::ptr::null(),
+            Some(urid) => urid,
+        };
+
+        let result = ::std::panic::catch_unwind(|| (&*(handle as *const Self)).unmap(urid));
+
+        match result {
+            Ok(Some(uri)) => uri.as_ptr(),
+            _ => ::std::ptr::null(), // FIXME: mapper panics should not be silenced
+        }
+    }
+
+    /// Create an unmap interface.
+    ///
+    /// This method clones the mapper and creates a self-contained `UnmapInterface`.
+    fn make_unmap_interface(&self) -> UnmapInterface<Self> {
+        let mut mapper = Box::pin(self.clone());
+        let mut unmap = Box::pin(sys::LV2_URID_Unmap {
+            handle: mapper.as_mut().get_mut() as *mut Self as *mut c_void,
+            unmap: Some(Self::extern_unmap),
+        });
+        let feature = Box::pin(core::sys::LV2_Feature {
+            URI: sys::LV2_URID__unmap.as_ptr() as *const c_char,
+            data: unmap.as_mut().get_mut() as *mut sys::LV2_URID_Unmap as *mut c_void,
+        });
+        UnmapInterface {
+            mapper,
+            unmap,
+            feature,
+        }
+    }
 }
 
 /// A simple URI → URID mapper, backed by a standard `HashMap` and a `Mutex` for multi-thread
 /// access.
-#[derive(Default)]
-pub struct HashURIDMapper(Mutex<HashMap<UriBuf, URID>>);
+#[derive(Default, Clone)]
+pub struct HashURIDMapper(Arc<Mutex<HashMap<UriBuf, URID>>>);
```

Comment:
Having a struct containing an Arc around its data and then implement clone on it is really unexpected behavior for users and should be avoided: when calling `clone()` on an object users expect it to be a deep clone, but here it makes just a simple reference count increment and still shares the same data.

`Rc` and `Arc` here should be created/used by the user explicitly, and only if needed (you can use `Borrow` if you want to be generic over it), see the [`Arc` docs](https://doc.rust-lang.org/std/sync/struct.Arc.html#cloning-references) and the associated [Clippy lint](https://rust-lang.github.io/rust-clippy/master/#clone_on_ref_ptr).